servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/deleteUser.js

```diff
@@ -0,0 +1,184 @@
+const { Roles } = require('@orbiting/backend-modules-auth')
+const { publishMonitor } = require('../../../../../lib/slack')
+
+const cancelPledge = require('./cancelPledge')
+const deleteStripeCustomer = require('../../../lib/payments/stripe/deleteCustomer')
+
+const deleteRelatedData = async (userId, pgdb) => {
+  // get all related tables
+  // https://stackoverflow.com/questions/5347050/sql-to-list-all-the-tables-that-reference-a-particular-column-in-a-table
+  const keepRelations = [ 'pledges', 'stripeCustomers' ]
+  const relations = await pgdb.query(`
+    select
+      R.TABLE_SCHEMA as schema,
+      R.TABLE_NAME as table
+    from INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE u
+    inner join INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS FK
+      on U.CONSTRAINT_CATALOG = FK.UNIQUE_CONSTRAINT_CATALOG
+      and U.CONSTRAINT_SCHEMA = FK.UNIQUE_CONSTRAINT_SCHEMA
+      and U.CONSTRAINT_NAME = FK.UNIQUE_CONSTRAINT_NAME
+    inner join INFORMATION_SCHEMA.KEY_COLUMN_USAGE R
+      on R.CONSTRAINT_CATALOG = FK.CONSTRAINT_CATALOG
+      and R.CONSTRAINT_SCHEMA = FK.CONSTRAINT_SCHEMA
+      and R.CONSTRAINT_NAME = FK.CONSTRAINT_NAME
+    where
+      U.TABLE_NAME = 'users'
+  `)
+    .then(rels => rels
+      .filter(rel => keepRelations.indexOf(rel.table) === -1)
+    )
+  relations.unshift({ // needs to be first
+    schema: 'public',
+    table: 'discussionPreferences'
+  })
+  relations.push({ // needs to be last
+    schema: 'public',
+    table: 'eventLog'
+  })
+  return Promise.all([
+    pgdb.query(`
+      DELETE
+        FROM sessions s
+      WHERE
+        ARRAY[(s.sess #>> '{passport, user}')::uuid] && :userIds
+    `, {
+      userIds: [ userId ]
+    }),
+    ...relations.map(rel =>
+      pgdb[rel.schema][rel.table].delete({
+        userId // assume the foreign column is always called userId
```

Comment:
Wondering if we should remove User ID from `comments.votes`; it could be construed as personal data but we do not need it to run our system.

Reply:
good catch! I didn't think about it. I'll pull it from all the votes arrays.